### PR TITLE
HPCC-28072 Add "DFS" option to Dali Admin

### PR DESCRIPTION
**Pull request by kunalaswani**
DaliAdmin implementation of DFSCheck, DFSExists, DFSLS

Signed-off-by: Kunal Aswani <Kunal.Aswani@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/components/DFSExists.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import { SizeMe } from "react-sizeme";
+import { csvParse } from "d3-dsv";
+import { DaliService, WsDali } from "@hpcc-js/comms";
+import { scopedLogger } from "@hpcc-js/util";
+import { TableGroup } from "./forms/Groups";
+import nlsHPCC from "src/nlsHPCC";
+import { HolyGrail } from "../layouts/HolyGrail";
+
+class MyDaliService extends DaliService {
+    DFSExists(request: WsDali.DFSExistsRequest): Promise<WsDali.ResultResponse> {
```
Comment (jeclrsg) [inline esp/src/src-react/components/DFSExists.tsx:12]: This extended version of DaliService could be removed. The function added exists on the service being imported from hpcc-js/comms (https://github.com/hpcc-systems/Visualization/blob/trunk/packages/comms/src/services/wsdl/WsDali/v1.04/WsDali.ts#L152)

File: esp/src/src-react/components/DFSLS.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import { SizeMe } from "react-sizeme";
+import { csvParse } from "d3-dsv";
+import { DaliService, WsDali } from "@hpcc-js/comms";
+import { scopedLogger } from "@hpcc-js/util";
+import { TableGroup } from "./forms/Groups";
+import nlsHPCC from "src/nlsHPCC";
+import { HolyGrail } from "../layouts/HolyGrail";
+
+class MyDaliService extends DaliService {
+    DFSLS(request: WsDali.DFSLSRequest): Promise<WsDali.ResultResponse> {
```
Comment (jeclrsg) [inline esp/src/src-react/components/DFSLS.tsx:12]: This extended version of DaliService could be removed. The function added exists on the service being imported from hpcc-js/comms (https://github.com/hpcc-systems/Visualization/blob/trunk/packages/comms/src/services/wsdl/WsDali/v1.04/WsDali.ts#L156)

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@kunalaswani I made a few comments about those extended MyDaliService objects. I think those can all be removed, because it looks like the functions being added already exist in the version in hpcc-js

**Review by jeclrsg — APPROVED** [review]

**Review by GordonSmith — COMMENTED** [review]
@kunalaswani @jeclrsg should this target 8.10.x?

**Comment by kunalaswani** [comment]
@GordonSmith I targeted master as this is a new feature for UI, but if you think it needs to be in 8.10 i can rebase

**Comment by GordonSmith** [comment]
@kunalaswani master is technically correct, but in general we are targeting 8.10.x for ECL Watch 9 changes - so for consistency can you retarget 8.10.x

**Comment by kunalaswani** [comment]
Please use this new PR. https://github.com/hpcc-systems/HPCC-Platform/pull/16697

**Pull request closed without merging**